rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -202,8 +202,14 @@ rmw_get_implementation_identifier()
 }
 
 rmw_ret_t
-rmw_init()
+rmw_init(const char * qos_xml_filename)
 {
+  std::string qos_xml_filename_str(qos_xml_filename);
+  if (!qos_xml_filename_str.empty()) {
+    RMW_SET_ERROR_MSG("rmw_opensplice does not support passing QoS profiles via XML files");
```

Comment:
Unfortunately, the only way of passing an XML file with the QoS profiles to OpenSplice is via the `QosProvider` API, which is only available in the `isocpp` variant. We're currently using `dcps`, which is incompatible with `isocpp`